File: sum_root_to_leaf_numbers.py
#!/usr/bin/env python

# Definition for a binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param root, a tree node
    # @return an integer
    def sumNumbers(self, root):
        return self.pathSum(root, 0)

    def pathSum(self, root, nums):
        if not root:
            return 0

        nums = nums*10 + root.val
        if not root.left and not root.right:
            return nums

        return self.pathSum(root.left, nums) + self.pathSum(root.right, nums)

File: leetcode/SumRoottoLeafNumbers.java
public class SumRoottoLeafNumbers {
    public int sumNumbers(TreeNode root) {
        return root == null ? 0 : pathSum(root, 0);
    }
    
    private int pathSum(TreeNode root, int sum) {
        if (root == null) {
            return 0;
        }

        sum = sum * 10 + root.val;
        if (root.left == null && root.right == null) {
            return sum;
        }

        return pathSum(root.left, sum) + pathSum(root.right, sum);
    }
}